### ISSUE TITLE
Help wanted?

### ISSUE THREAD

**Issue by SubatomicPlanets** [batschigames/AVeryPotterVRGame#1]
Hey, I am a game developer and just wanted to ask if I can help on this game? I am currently making my own VR game, and I think we could sort of use the best of both worlds to create one magnificent game! 

I have called my game HogwartsVR but I didn't publish it and I don't plan on doing that. But while working on it I realized that a small team would be way more fun than if I am doing it alone... 

I am more of a programmer than artist, but I can also do basic 3d modelling. Some of the features my game currently has are these:
- multiplayer (using netcode for gameobjects)
- voice chat (my own custom solution, encrypted and compressed but there is no echo cancellation yet)
- optional AI NPCs (using mistral 7b and custom voices using piper TTS)
- open world (all of Hogwarts and grounds)
- small game size (just 200mb)
- cross platform (windows, macos, android)
- VR is optional but recommended
- local splitscreen multiplayer 
- speech recognition using pocketsphinx
- no loading screens at all
- a character customizer 
- quidditch (works perfectly in multiplayer)
- gamepad support 
- great performance (runs on standalone)
- and more...

As you can see I have already got quite a lot. I started working on it last year. I just think it's sort of stupid that we are both making separate games individually and not one big game together... 

I guess I am just a bit lonely...
Please let me know if you want to contribute to my project or if you allow me to contribute to your project or if we can together make a new project. 

Thanks for reading this...

**Comment by batschigames** [comment]
Thanks so much for reaching out! I'm for sure interested in incorporating
your work into my existing game if that's something your interested in. I'm
impressed with all you have done so far, especially getting multiplayer
working! That is my most requested feature that I know is going to be a ton
of work.

I'll admit that I'm not able to work on my project as much as I'd like and
I agree working together would be fun!

I would love to know a bit more about what you have started. Is it in
unity? Is there a way I might be able to play it a little?

Just so you know vision. In order to avoid my game being taken down, I'm
trying to avoid any copyrighted names of anything in harry potter and I'm
not planning on making any money from it. (Let me know if you might have
different ideas than this).

Anyways... Let me know how you might invision moving forward.

**Comment by SubatomicPlanets** [comment]
Yes, my game is made with unity. I know about the copyright issues. It's a bit stupid having to rename all of the characters and locations, but I guess that's the only thing you can do to make it work, so yeah, I am okay with that. I also think it should be free, and I don't want any money either. 
I can maybe somehow send you the game, or the unity project, but I will first have to remove some stuff. I just made the game for me and my friends, so there is some personal stuff in there (Like the moving paintings which are videos of me...).
I think maybe the best thing for me to do is to just contribute most of my work towards your project but also continue my personal project. I think that makes the most sense.

**Comment by SubatomicPlanets** [comment]
I can send you a few screenshots from my phone version for now (ignore the white squares, they are the buttons to move around and pick stuff up): 
![Screenshot_20240610-040753_HogwartsVR](https://github.com/batschigames/AVeryPotterVRGame/assets/133254461/1b4ea4d5-69d7-4c55-8ce7-0e9f631489fd)
![Screenshot_20240610-040748_HogwartsVR](https://github.com/batschigames/AVeryPotterVRGame/assets/133254461/084a24b2-ca0f-45ba-ba76-bc3caab9f58e)
![Screenshot_20240610-040920_HogwartsVR](https://github.com/batschigames/AVeryPotterVRGame/assets/133254461/7a8f6bb0-2ee3-4cdd-9fe3-18626bf1488a)
![Screenshot_20240610-041054_HogwartsVR](https://github.com/batschigames/AVeryPotterVRGame/assets/133254461/0885d82d-627b-4a17-bed4-b88fae6a3023)

**Comment by batschigames** [comment]
Yeah that sounds great. When I have a second I can try to get you access to
my unity project. It is pretty huge so I'll have to search for the best way
to do that...

**Comment by SubatomicPlanets** [comment]
Okay, I have removed the paintings and some other stuff. I can send you the game using a public Google Drive link. I have uploaded a Windows version for Quest Link and an Android version for Quest standalone. One important thing to note is that it only supports meta quest 2 and 3. I don't think other VR headsets will work. You can also play the PC version without any VR headset at all, or use the APK on your phone. 

Most of the castle and grounds are quite empty, and the graphics are not so great, but as I said, I am more of a programmer than a 3D artist. The spells that currently work are:
- Lumos
- Nox
- Accio
- Depulso
- Engorgio
- Reducio
- Tarantallegra
- Avada Kedavra (only works on some NPCs and is not finished yet)

I personally think that the broom mechanics in my game are better than in yours, but that's just my opinion. I think it might make some people even more motion sick than yours... Just try it out by grabbing a broom in the great hall or broom shed and "getting onto it"

Somewhere there is a white painting which is the fat lady. Just go up to it and say "pig snout" to enter. In the dungeons, there is a wall, and behind it is the Slytherin common room. The password for that is "pure-blood" (but it's very hard to find...).

Also, before playing, go into the settings tab in the menu and set the quality to low if it's on standalone and to medium if it's on PC VR. High just turns on shadows and a bit of other stuff. Do not touch the "third person" and "half resolution" settings. They only work when not playing in VR. They should be set to off (third person actually works in VR and it's weird) The "hide character" setting makes your body invisible. The height slider currently does nothing at all.

And lastly, the performance is not great on Quest 2 standalone, but in some places it is fine. The worst lag happens when looking at the castle from far away.

Here is the link:
https://drive.google.com/drive/folders/1mKSfANKlsjQSNHlAG2oeOzr9Fu2WbfaY?usp=drive_link

Let me know what you think, and if you can share your project. How big is it exactly (without the library folder, since unity auto generates it when you start it)?
I think the best way would be to upload it to github in either a public repo so that it's open source, or in a private repo where you can add me as a collaborator. But I'm not sure about file size limits on GitHub and so on, so it's up to you.
Thanks!

**Comment by batschigames** [comment]
Will you send me your email address? I'm going to try to send you my project

**Comment by SubatomicPlanets** [comment]
Okay, thanks for sending it! I will just look at it for now and I will contact you again when I have something to add to the game or so.

**Comment by batschigames** [comment]
Awesome :) I think adding in a multiplayer would be really cool but I know
it can be really tricky... feel free to email me at ***@***.***
in the future. Have fun!
